### ISSUE TITLE
Approach of disabling 'Compare with Gatsby' button when no option is selected [jamstack.js] [cms.js] 

### ISSUE THREAD

**Issue by qmixi** [gatsbyjs/gatsby#19127]
## Description

`Compare with Gatsby` button in `jamstack` and `cms` pages uses Gatsby `<Link/>` component under the hood. It might be reasonable since it lets the user navigate to a different url but on the other hand its functionality depends on JavaScript.

The usability problem is that currently, the user can click on the button even though no option is selected and the page is re-reloaded then. Unfortunately, anchor tag doesn't have natively supported `disabled` prop, so the only one way to disable this `<a/>` tag is to prepare custom CSS class with `pointer-events: none` property.

Because adding the ability to disable this button could boost the UX of this feature significantly,  my suggestion in here is to use `<Button/>` component instead of `<Link/>` which will allow us to disable it natively and use Gatsby `navigate` helper function to handle redirection properly on button click.

What do you think about this idea guys? I can prepare a PR with appropriate changes :)

<img width="932" alt="Screenshot 2019-10-29 at 22 39 00" src="https://user-images.githubusercontent.com/16977412/67811422-04de5580-fa9d-11e9-886a-f79caf8574c8.png">
<img width="959" alt="Screenshot 2019-10-29 at 22 39 27" src="https://user-images.githubusercontent.com/16977412/67811423-04de5580-fa9d-11e9-8db8-25a1b6a6bfd4.png">
<img width="1026" alt="Screenshot 2019-10-29 at 22 39 23" src="https://user-images.githubusercontent.com/16977412/67811424-04de5580-fa9d-11e9-94fc-b01987540df3.png">
<img width="1042" alt="Screenshot 2019-10-29 at 22 39 06" src="https://user-images.githubusercontent.com/16977412/67811425-0576ec00-fa9d-11e9-845b-fe9e4bb2617c.png">

(Screens above contain changes from PR https://github.com/gatsbyjs/gatsby/pull/19063 fixing problem with paddings)

**Comment by fk** [comment]
Hey @qmixi! 👋 Sorry for only managing now to get to this, and thanks for the thorough explanation!

After briefly looking at e.g. https://www.gatsbyjs.org/features/jamstack/, doing this makes sense to me. The existing [button component](https://github.com/gatsbyjs/gatsby/blob/bba2c2f52722278941e45e047fbd6025cef0bd66/www/src/components/button.js) might have all we need.

Let me ping @gillkyle though, too, who is much more familiar with the current implementation.

**Comment by gillkyle** [comment]
I'm embarrassed I didn't see that I got mentioned here sooner 🙈 so sorry! 

The code for these pages have been moved over under the gatsbyjs.com wing with our [unified site](https://www.gatsbyjs.com/blog/announcing-unified-gatsby/), so the changes would no longer be in this repo. I can open an internal ticket to revisit this behavior since I believe these pages are ones we'd like to revisit soon anyway. Thanks for raising this issue and the great investigation that will make providing details about it easy!
